s3backgrounddelete/s3backgrounddelete/IEMutil.py

```
@@ -49,5 +52,32 @@ def __init__(self, loglevel, eventcode, eventstring):
         self.log_iem()
 
     def log_iem(self):
-        IEM = "IEC:%sS%s:%s" % (self.severity, self.eventCode, self.eventString)
-        syslog.syslog(IEM)
+        self.send(self.module, self.eventCode, self.severity, self.eventString)
+
+    @classmethod
+    def send(self, module, event_id, severity, message_blob):
+        """Send the message."""
+        source = 'S'
+        component= 'S3'
+        EventMessage.init(component, source)
+
+        try:
+            EventMessage.send(module=module, event_id=event_id, severity=severity, message_blob=message_blob)
+        except Exception as exception:
+            msg = ("iem except:%s %s") % (
+                    exception, traceback.format_exc())
+            return False, msg
+        return True, None
+
+    @classmethod
+    def receive(self, component):
+        """Receive the incoming message."""
+        EventMessage.subscribe(component)
+
+        try:
+            alert = EventMessage.receive()
+        except Exception as exception:
+            msg = ("iem except:%s %s") % (
```

No need to return msg. 
see of we can log exception and continue..

---

Ignore exception but add a log message
Take logger as input to constructor.

---

done